### PR TITLE
Tela de campeões

### PR DESCRIPTION
**Pull request by cesar1287**

### CODE REVIEW

File: champions/src/main/java/com/github/champions/repository/ChampionsRepository.kt [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package com.github.champions.repository
+
+import com.github.lol4fun.core.api.Resource
+import com.github.lol4fun.core.repository.BaseRepository
```
Comment (valmirt) [inline champions/src/main/java/com/github/champions/repository/ChampionsRepository.kt]: Atualizar import do BaseRepository `import com.github.lol4fun.core.base.BaseRepository`

File: champions/src/main/java/com/github/champions/features/champions/business/ChampionsBusiness.kt [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import com.github.lol4fun.core.api.Status
+import com.github.lol4fun.core.model.Champions
+import com.github.lol4fun.extensions.serializeToMap
+import com.github.lol4fun.extensions.toChampionsDTO
+
+class ChampionsBusiness(
+    private val championsListener: ChampionsListener
+) {
+
+    private val repository: ChampionsRepository by lazy { ChampionsRepository() }
```
[the remark this answers is not in the document]
Comment (cesar1287) [reply champions/src/main/java/com/github/champions/features/champions/business/ChampionsBusiness.kt]: done

File: champions/src/main/java/com/github/champions/features/champions/view/ChampionsFragment.kt [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+import com.github.champions.R
+import com.github.champions.adapter.ChampionsAdapter
+import com.github.champions.features.champions.viewmodel.ChampionsViewModel
+import com.github.lol4fun.extensions.showSnackBar
+import com.github.lol4fun.util.ConstantsUtil.Champions.NUMBER_OF_COLUMNS_GRID_CHAMPIONS
+import kotlinx.android.synthetic.main.fragment_champions.*
+
+class ChampionsFragment: Fragment() {
+
+    private lateinit var viewModel: ChampionsViewModel
```
[the remark this answers is not in the document]
Comment (cesar1287) [reply champions/src/main/java/com/github/champions/features/champions/view/ChampionsFragment.kt]: done

File: champions/src/main/java/com/github/champions/features/champions/viewmodel/ChampionsViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.github.champions.features.champions.business.ChampionsBusiness
+import com.github.champions.features.champions.listener.ChampionsListener
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+
+class ChampionsViewModel : ViewModel(), ChampionsListener {
+
+    private val business: ChampionsBusiness by lazy { ChampionsBusiness(this) }
```
[the remark this answers is not in the document]
Comment (cesar1287) [reply champions/src/main/java/com/github/champions/features/champions/viewmodel/ChampionsViewModel.kt]: done

File: champions/src/main/java/com/github/champions/repository/ChampionsRepository.kt (hunk 1 above)
Comment (cesar1287) [reply champions/src/main/java/com/github/champions/repository/ChampionsRepository.kt]: done

**Pull request merged**